### PR TITLE
Add github action to assign maintainers to PRs

### PR DESCRIPTION
**Pull request by MarcelMue**
Signed-off-by: Marcel Mueller <marcel.mueller1@rwth-aachen.de>

### CODE REVIEW

**Review by cndoit18 — APPROVED** [review]

**Review by OrlinVasilev — APPROVED** [review]
looks great to me :)

File: .github/auto-assignees.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - yxxhero
+
+options:
+  ignore_draft: true
+  ignored_keywords:
+    - WIP
+    - wip
+    - DO NOT MERGE
+  enable_group_assignment: true
+  number_of_reviewers: 3
```
Comment (bitsf) [inline .github/auto-assignees.yaml:30]: I think 2 is enough

File: .github/auto-assignees.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# This assigns a PR to its author
+addAssignees: true
```
Comment (bitsf) [inline .github/auto-assignees.yaml:2]: this seems not required currently

Comment (MarcelMue) [reply .github/auto-assignees.yaml:2]: Why do you think so? Every example (even the upstream one) explicitly sets this to true:

https://github.com/kentaro-m/auto-assign-action#single-reviewers-list

And matches what `harbor` repo has set:
https://github.com/goharbor/harbor/blob/a826720cc94dd9ac8d3495ba74a913055a2a9268/.github/auto-assignees.yml#L3

File: .github/auto-assignees.yaml (hunk 1 above)
Comment (MarcelMue) [reply .github/auto-assignees.yaml]: Changed.

File: .github/auto-assignees.yaml (hunk 2 above)
Comment (OrlinVasilev) [reply .github/auto-assignees.yaml:2]: it should be either true or author, we need true so we have array of assignees

Comment (bitsf) [reply .github/auto-assignees.yaml:2]: please add wy65701436 , and assign to him

Comment (MarcelMue) [reply .github/auto-assignees.yaml:2]: Do you mean adding @wy65701436 to the list of assignees?

I took the current list from here: https://github.com/goharbor/harbor-operator/blob/master/MAINTAINERS

Comment (OrlinVasilev) [reply .github/auto-assignees.yaml:2]: @MarcelMue correct but Yan Wang @wy65701436 is currently the overall tech lead for Harbor so it's good to have him :)

Comment (OrlinVasilev) [reply .github/auto-assignees.yaml:2]: @wy65701436 can you confirm please

Comment (MarcelMue) [reply .github/auto-assignees.yaml:2]: Done.

**Comment by OrlinVasilev** [comment]
I think that's is done shall we merge ? @goharbor/all-maintainers

**Review by Vad1mo — APPROVED** [review]

**Pull request merged**